Dockerfile

```diff
@@ -1,32 +1,41 @@
-FROM ocaml/opam:alpine-3.20-ocaml-5.2 as build
+FROM ocaml/opam:alpine-3.20-ocaml-5.2 AS build
 
 # Install system dependencies
-RUN sudo apk update && sudo apk add --update libev-dev openssl-dev gmp-dev oniguruma-dev inotify-tools curl-dev autoconf
+RUN apk -U upgrade --no-cache && apk add --no-cache \
+    autoconf \
+    curl-dev \
+    gmp-dev \
+    inotify-tools \
+    libev-dev \
+    oniguruma-dev \
+    openssl-dev
 
 # Branch freeze was opam-repo HEAD at the time of commit
-RUN cd opam-repository && git pull origin c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
+RUN cd ~/opam-repository && git fetch -q origin master && git reset --hard c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
 
 WORKDIR /home/opam
 
-# Install Opam dependencies
-ADD ocamlorg.opam ocamlorg.opam
+# Install opam dependencies
+COPY --chown=opam --link ocamlorg.opam .
```

Comment:
```suggestion
COPY --chown=opam ocamlorg.opam .
```

Reply:
Pure misclick, from my phone, in the train. Total fumble